pagination/helpers.go

```diff
@@ -87,3 +102,60 @@ func ParsePageTokenToTime(token string) (time.Time, error) {
 	}
 	return result, err
 }
+
+// GetNextPageTokenFromTime generates a page token. This function should be used when generating a next page token
+// in a List operation on a time.Time field.
+func GetNextPageTokenFromTime(updatedAt time.Time) string {
+	if updatedAt.IsZero() {
+		return ""
+	}
+	return NewPageToken(updatedAt)
+}
+
+// SetCurrentPage generates a set of repository.Option to retrieve results for a specific page.
+func SetCurrentPage(opts []repository.Option, p Pagination) ([]repository.Option, error) {
+	opts = append(opts, firestore.OrderBy(firestore.Ascending("updated_at")))
+	opts, err := setMaxResults(opts, p)
+	if err != nil {
+		return nil, err
+	}
+	if p == nil || len(p.GetPageToken()) == 0 {
+		return opts, nil
+	}
+	updatedAt, err := ParsePageTokenToTime(p.GetPageToken())
+	if err != nil {
+		return nil, err
+	}
+	opts = append(opts, firestore.StartAt(updatedAt))
+	return opts, nil
+}
+
+// setMaxResults establishes the max number of items that should be returned from firestore.
+// This function includes an extra element in the MaxResults option, this last element is used for pagination.
+// The last element should be discarded before the list is returned to the user.
+// See GetListAndCursor for more information.
+func setMaxResults(opts []repository.Option, sg pageSizeGetter) ([]repository.Option, error) {
+	p := PageSize(sg)
+	if p == InvalidValue {
+		return nil, errors.New("invalid page size")
+	}
+	return append(opts, firestore.MaxResults(int(p)+1)), nil
+}
```

Comment:
These two are specific to Firebase. Let's please move them to `repository/firestore/options.go`. You can create a separate file like`options_pagination.go` inside that package if you want.

Reply:
Let's also please leave the appending up to the application to be consistent with other option functions.